clang/include/clang/Sema/Sema.h

```diff
@@ -240,7 +240,43 @@ class PossiblyUnreachableDiag;
 class RISCVIntrinsicManager;
 class SemaPPCallbacks;
 class TemplateDeductionInfo;
-
+class EnumVectorManager{
+  public:
+    std::vector<std::string> EnumVector;
+};
+class EnumNameVisitor {
+private:
+  std::string EnumName;
+public:
+  void VisitDeclRefExpr(DeclRefExpr *Node) {
+    const Decl *D = Node->getDecl();
+    // Check if the declaration referenced is an enum constant
+    if (const auto *ECD = dyn_cast<EnumConstantDecl>(D)) {
+      if (const auto *ED = dyn_cast<EnumDecl>(ECD->getDeclContext())) {
+        // Construct the fully qualified name
+        EnumName = ED->getName().str() + "::" + ECD->getName().str();
+        return; // No error for valid enum constants
+      }
+    }
+  }
+  void VisitExpr(Expr *Ex) {
+    if (!Ex->children().empty()) {
+      // Recursively visit child expressions
+      for (auto *Child : Ex->children()) {
+        if (auto *ChildExpr = dyn_cast<Expr>(Child)) {
+          VisitExpr(ChildExpr);
+        }
+      }
+    }
+    // Now handle the current expression node
+    if (auto *DRE = dyn_cast<DeclRefExpr>(Ex)) {
+      VisitDeclRefExpr(DRE); // If it's a DeclRefExpr, visit it
+      return;
+    }
+  }
+  // Method to retrieve the last fetched string
+  StringRef getEnumName() const { return StringRef(EnumName); }
+};
 } // namespace sema
```

Comment:
Move the class definition into another file (Sema.cpp). Notice where the classes mentioned above are defined.